File: c++/leetcode/1464. Maximum Product of Two Elements in an Array.cpp
class Solution {
public:
    int maxProduct(vector<int>& nums) {
        int first = 0;
        int second = 0;
        
        const auto find_two = [&](int& a) {
            if(a > first) 
                second = exchange(first, a);
            else
                second = max(second, a);
        };
        
        
        for_each(nums.begin(), nums.end(), find_two);
        
        
        return --first * --second;
    }
}; 

File: javascript/leetcode/1464. Maximum Product of Two Elements in an Array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxProduct = function(nums) {
    let first = 0, second = 0
    
    nums.forEach(el => {
        if(el > first)
            second = first, first = el
        else
            second = Math.max(second, el)
    })
    
    return --first * --second
};
